### PR TITLE
homework 5

### PR DESCRIPTION
**Pull request by dantarasov**
homework 5

### CODE REVIEW

File: integration/__specs__/tests/analytics/deleteItems.spec.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    });
+
+    afterAll(() => {
+        cartPage.destroy();
+        mock.close();
+    });
+
+    test('Check events in dataLayer after creating & deleting', async () => {
+        await cartPage.fulfill();
+        await cartPage.addCartItem();
```
Comment (LMikhail1) [inline integration/__specs__/tests/analytics/deleteItems.spec.ts:29]: Where are these methods?

File: integration/__specs__/tests/analytics/deleteItems.spec.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        reporter.startStep('Modal window opened');
+        const openEvent = await waitForDataLayer({ name: 'FormInteraction' });
+        expect(openEvent).toStrictEqual({ name: 'FormInteraction', value: 'Open' });
+        reporter.endStep();
+
+        const cartModal = await cartPage.getCartModal();
+
+        const nameInput = await cartModal.getCartName();
+        const priceInput = await cartModal.getCartPrice();
+        const quantityInput = await cartModal.getCartQuantity();
```
Comment (LMikhail1) [inline integration/__specs__/tests/analytics/deleteItems.spec.ts:40]: Where are these methods?

File: integration/__specs__/tests/analytics/deleteItems.spec.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
+import { Mock } from '@Core/mock';
+import { CartPage } from '@Components/cartPage/cartPage/cartPage';
+import { GetCartItemsMock } from '@Mocks/api/mockio/v1/id/get';
+import { faker } from '@faker-js/faker';
```
Comment (LMikhail1) [inline integration/__specs__/tests/analytics/deleteItems.spec.ts:4]: Where do you get faker from?

Comment (dantarasov) [reply integration/__specs__/tests/analytics/deleteItems.spec.ts:4]: From '@faker-js/faker'. I just installed it by npm

Comment (LMikhail1) [reply integration/__specs__/tests/analytics/deleteItems.spec.ts:4]: but I don't see it in package.json

Comment (dantarasov) [reply integration/__specs__/tests/analytics/deleteItems.spec.ts:4]: maybe this is because flag --save-dev

Comment (LMikhail1) [reply integration/__specs__/tests/analytics/deleteItems.spec.ts:4]: It is better to install it also in the project

File: integration/__specs__/tests/analytics/deleteItems.spec.ts [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        expect(valueOfPrice).toStrictEqual(data.price);
+        expect(valueOfQuantity).toStrictEqual(data.quantity);
+        reporter.endStep();
+
+        reporter.startStep('Check that a length of the cartList array has incremented by 1');
+        expect(items.length).toBe(4);
+        reporter.endStep();
+
+        const [item] = await list.getCartItems();
+        item.deleteCartItem();
```
Comment (LMikhail1) [inline integration/__specs__/tests/analytics/deleteItems.spec.ts]: Isn't this method asynchronous ?

**Review by LMikhail1 — APPROVED** [review]

**Pull request merged**